qiling/os/posix/syscall/wait.py

```diff
@@ -12,11 +12,18 @@
 from qiling.os.posix.const_mapping import *
 from qiling.exception import *
 from qiling.utils import *
+from qiling.os.posix.const import ECHILD
+
 
 def ql_syscall_wait4(ql, wait4_pid, wait4_wstatus, wait4_options, wait4_rusage, *args, **kw):
-    spid, status, rusage = os.wait4(wait4_pid, wait4_options)
-    if wait4_wstatus != 0:
-        ql.mem.write(wait4_wstatus, ql.pack32(status))
-    regreturn = spid
+    wait4_pid = ql.unpacks(ql.pack(wait4_pid))  # convert to signed
+    try:
+        spid, status, rusage = os.wait4(wait4_pid, wait4_options)
+        if wait4_wstatus != 0:
+            ql.mem.write(wait4_wstatus, ql.pack32(status))
+        regreturn = spid
+    except ChildProcessError:
+        regreturn = -ECHILD
     logging.info("wait4(%d, %d) = %d"% (wait4_pid, wait4_options, regreturn))
-    return regreturn
+    ql.os.definesyscall_return(regreturn)
```

Comment:
the new method just return. it handling will be done after the return.

Reply:
That is convenient. I fixed it now.